rpcs3/Emu/Cell/RawSPUThread.cpp

```diff
@@ -116,6 +161,44 @@ bool RawSPUThread::write_reg(const u32 addr, const u32 value)
 
 	switch (offset)
 	{
+		
+	case MFC_MSSync_offs:
+	{
+		bool q = false;
+		
+		if (MSSync_reg_left)
+		{
+			while (MSSync_reg_left)
+			{
+				if (mfc_queue[0].MSSync_type == 2)
+				{
+					do_dma_transfer(mfc_proxy[0]);
+					mfc_queue.end_pop();
+					if (mfc_queue.size() == 15)
+					{
+						ch_event_stat |= SPU_EVENT_QV;
+					}
+				}
+			}
+			
+			q = true;
+		}
```

Comment:
end_pop can only be used by mfc side. ur transfering mfc_proxy but checking mfc_queue? Is that intentional?
Also is there any way to make the mfc_sync request using a spu_channel_t? As is you have a while loop with no guaranteed break or emu.stopped check which will potentially lock the emulator, using `read_channel` would avoid this and possibly fix the need for the double loop.